core/src/main/java/hudson/util/TextFile.java

```diff
@@ -50,6 +51,70 @@
 public class TextFile {
     public final File file;
 
+    /**
+     * Represents a stream over the lines of the text file.
+     * <p>
+     * Although <code>LinesStream</code> implements {@link java.lang.Iterable},
+     * it is intended to be first used to initialize a resource in a try-with-resources
+     * statement and then iterated, as in:
+     * <pre>
+     *  try (LinesStream stream = lines()) {
+     *      for (String line : stream) {
+     *          ...
+     *      }
+     *  }
+     * </pre>
+     * This pattern ensures that the underlying file handle is closed properly.
+     * <p>
+     * Like {@link java.nio.file.DirectoryStream}, <code>LinesStream</code>
+     * supports creating at most one <code>Iterator</code>. Invoking
+     * {@link #iterator()} to obtain a second or subsequent <code>Iterator</code>
+     * throws <code>IllegalStateException</code>.
+     */
+    public class LinesStream implements Closeable, Iterable<String> {
```

Comment:
`@since TODO` or make it `@Restricted` if there is no plan to use it externally

Reply:
Maybe you want to create a new utility class and pass file as a constructor parameter. There are other use-cases for such class, so no need to keep it nested

Reply:
I extracted the class to top-level and added `@since TODO`.